pandas/tseries/index.py

```diff
@@ -833,7 +833,7 @@ def astype(self, dtype):
         if dtype == np.object_:
             return self.asobject
         elif dtype == _INT64_DTYPE:
```

Comment:
So, now `.astype(x)` for `x in ['int32', 'int16', 'int8']` converts `NaN` to 0. Is it OK? And should I add some tests for these as well?